### PR TITLE
UIIN-1920 Implement Advanced Search modal

### PR DESCRIPTION
**Pull request by BogdanDenis**
## Description
Added Advanced Search to Inventory Search

## Screenshots

https://github.com/folio-org/ui-inventory/assets/19309423/14c3369f-894f-4923-a7e9-bd69f8de8bfb

## Issues
[UIIN-1920](https://issues.folio.org/browse/UIIN-1920)

## Related PRs
https://github.com/folio-org/stripes-smart-components/pull/1376
https://github.com/folio-org/stripes-components/pull/2091

### CODE REVIEW

**Review by mariia-aloshyna — APPROVED** [review]
Wow, it's a kinda cool feature! LGTM

File: src/utils.js [hunk 1]
```diff
@@ -268,6 +271,37 @@ export function getIsbnIssnTemplate(queryTemplate, identifierTypes, queryIndex)
...
+  const queryIndex = queryParams?.qindex ?? 'all';
+  const queryValue = get(queryParams, 'query', '');
+  const queryMatch = get(queryParams, 'match', 'exactPhrase');
+  let queryTemplate = getAdvancedSearchQueryTemplate(queryIndex, queryMatch);
+
+  if (queryIndex.match(/isbn|issn/)) {
+    // eslint-disable-next-line camelcase
+    const identifierTypes = resourceData?.identifier_types?.records ?? [];
+    queryTemplate = getIsbnIssnTemplate(queryTemplate, identifierTypes, queryIndex);
+  }
```
Comment (Dmytro-Melnyshyn) [inline src/utils.js:286]: It looks like it's redundant.

File: src/utils.js [hunk 2]
```diff
@@ -268,6 +271,37 @@ export function getIsbnIssnTemplate(queryTemplate, identifierTypes, queryIndex)
   return template(queryTemplate)({ identifierTypeId });
 }
 
+const getAdvancedSearchQueryTemplate = (queryIndex, matchOption) => fieldSearchConfigurations[queryIndex]?.[matchOption];
+
+export function buildAdvancedSearchQuery(queryParams, pathComponents, resourceData, logger, props) {
```
Comment (Dmytro-Melnyshyn) [inline src/utils.js:276]: Can we utilize the `buildQuery` function from the `builtManifestObject.js` file? By simply adding an `if (queryIndex === AdvancedSearch)` condition?

**Review by Dmytro-Melnyshyn — CHANGES_REQUESTED** [review]

**Comment by Dmytro-Melnyshyn** [comment]
1. Seems the `Query` search option should be excluded from the advanced search.
2. `Contans all` is changed to `Exact phrase` after the second opening of the advanced search modal.
3. The request comprises escapes that lead to incorrect results number `(keyword==/string \"a\" or title all \"b\") sortby title`.
4. [Page falls](https://github.com/folio-org/ui-inventory/assets/77053927/4f802011-d416-4151-88b7-e8a33f0e74f0)
5. Page falls with the same error if I select the `Advanced search` search option and paste `(keyword==/string "a" or title all "b") sortby title`.

**Comment by Dmytro-Melnyshyn** [comment]
> 2. I haven't found a solution how to get used match option 100% of the time. It's difficult because match option is not a single symbol, but a combination of `==`, `==/string`, `all` etc and prefix or postfix `*` symbol
> 3. I think that's the correct format according to docs https://github.com/folio-org/mod-search#cql-query-modifiers

2. Have you tried the solution from this [Spike](https://wiki.folio.org/pages/viewpage.action?pageId=118265036)?
3. When quotes are escaped, they are treated as user input. I think the request query should be `(keyword all "a" and oclc="*b*") sortby title` instead of `(keyword all \"a\" and oclc=\"*b*\") sortby title`.

File: src/utils.js [hunk 3]
```diff
@@ -22,6 +22,7 @@ import {
 import moment from 'moment';
 
 import { FormattedUTCDate } from '@folio/stripes/components';
+import { makeQueryFunction } from '@folio/stripes/smart-components';
```
Comment (Dmytro-Melnyshyn) [inline src/utils.js:25]: Please fix eslint warnings in this file.

**Review by Dmytro-Melnyshyn — CHANGES_REQUESTED** [review]

**Comment by Dmytro-Melnyshyn** [comment]
[Page crashes](https://github.com/folio-org/ui-inventory/assets/77053927/4b75c29a-271a-48cb-a7d7-9328129b820e)

**Pull request merged**